packages/cache/src/CacheStorage.ts

```diff
@@ -1,37 +1,62 @@
 import * as fs from "fs-extra";
+import * as path from "path";
 import { logger } from "backfill-logger";
+import { outputFolderAsArray } from "backfill-config";
 
 export interface ICacheStorage {
-  fetch: (hash: string, destinationFolder: string) => Promise<Boolean>;
-  put: (hash: string, sourceFolder: string) => Promise<void>;
+  fetch: (
+    hash: string,
+    destinationFolder: string | string[]
```

Comment:
This name is not accurate anymore since it is singular... it may be confusing... or it may not be... I don't know